### PR TITLE
Allow client interceptor to send headers

### PR DESCRIPTION
**Pull request by mairbek**

### CODE REVIEW

File: test/grpc/integration/interceptor_headers.exs [hunk 1]
```diff
@@ -0,0 +1,60 @@
+defmodule GRPC.Integration.EndpointTest do
```
Comment (tony612) [inline test/grpc/integration/interceptor_headers.exs]: Could you rename this file to `interceptor_headers_test.exs` and rename the module to `GRPC.Integration.InterceptorHeadersTest`? btw, it's better to rename them to "client_interceptor..."

File: test/grpc/integration/interceptor_headers.exs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    def init(_), do: []
+
+    def call(req, stream, next, _) do
+      headers = GRPC.Stream.get_headers(stream)
+      %{name: original_name} = req
+      label = headers["x-test-label"]
+      updated = Helloworld.HelloRequest.new(name: "#{original_name} #{label}")
+      next.(updated, stream)
+    end
+  end
```
Comment (tony612) [inline test/grpc/integration/interceptor_headers.exs]: Why is this interceptor is needed? I guess we only need verify headers in HelloServer?

File: lib/grpc/transport/http2.ex [hunk 3]
```diff
@@ -43,6 +43,7 @@ defmodule GRPC.Transport.HTTP2 do
     |> append_custom_metadata(stream.channel.headers)
     |> append_encoding(opts[:grpc_encoding])
     |> append_timeout(opts[:timeout])
+    |> append_stream_metadata(stream)
```
Comment (tony612) [inline lib/grpc/transport/http2.ex]: Could we use `append_custom_metadata(stream.headers)`?

**Comment by mairbek** [comment]
thank you @tony612 , i have addressed the comments

**Pull request merged**